### PR TITLE
Allow opening deeplinks with custom schema

### PR DESCRIPTION
**Pull request by jokakilla**
Before this change opening URLs with a schema different from http/https wasn't possible because the code always prepended http.
With this change http is only prepended when no schema is present.

### CODE REVIEW

**Comment by jokakilla** [comment]
https://github.com/SecUSo/privacy-friendly-qr-scanner/issues/224

File: app/src/main/java/com/secuso/privacyfriendlycodescanner/qrscanner/ui/resultfragments/URLResultFragment.java [hunk 1]
```diff
@@ -95,23 +95,14 @@ public void onProceedPressed(Context context) {
...
-                caption = getResources().getStringArray(R.array.url_array)[0];
-                startActivity(Intent.createChooser(url, caption));
-            } else {
-                Intent url = new Intent(Intent.ACTION_VIEW);/// !!!!
-                url.setData(Uri.parse(qrurl).normalizeScheme());
-                caption = getResources().getStringArray(R.array.url_array)[0];
-                startActivity(Intent.createChooser(url, caption));
-
+            String urlForIntentData = qrurl;
+            if (!qrurl.toLowerCase().contains("://")) {
```
Comment (coderPaddyS) [inline app/src/main/java/com/secuso/privacyfriendlycodescanner/qrscanner/ui/resultfragments/URLResultFragment.java:99]: This does not respect the RFC 3986 which specifies what a valid protocol-scheme looks like.
A regex to match that should be `^[a-zA-Z][a-zA-Z0-9+.-]*:.*$`, but this needs to be checked and tested.

**Review by coderPaddyS — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/secuso/privacyfriendlycodescanner/qrscanner/ui/resultfragments/URLResultFragment.java (hunk 1 above)
Comment (coderPaddyS) [reply app/src/main/java/com/secuso/privacyfriendlycodescanner/qrscanner/ui/resultfragments/URLResultFragment.java]: Thanks for improving on this! :smile: 
I'll take a look and test your changes myself in the next few days and will check if using https as default won't break any expected use case.
We'd appreciate if you'd rebase your changes onto `development` and target `development` for the PR.

**Comment by jokakilla** [comment]
Any chance this change will be merged? Should I do any further adaptions?

**Review by coderPaddyS — APPROVED** [review]

**Comment by coderPaddyS** [comment]
I'm sorry for taking this long.
If looked into branch and noticed that a rebase is not necessary as no merge conflicts happen.
Thank you for your contribution :D

**Pull request merged**